geoapps/octree_creation/driver.py

```diff
@@ -116,6 +120,7 @@ def refine_tree_from_curve(
         mesh: TreeMesh,
         curve: Curve,
         levels: list[int] | np.ndarray,
+        diagonal_balance: bool,
```

Comment:
Defaulting to True here too is fine. Looks like you are already using a kwarg later in the code anyway.